src/sentry/static/sentry/app/components/forms/formField.jsx

```diff
@@ -31,22 +31,27 @@ export default class FormField extends React.Component {
   };
 
   constructor(props, context) {
-    super(props);
+    super(props, context);
+    // XXX(dcramer): defer the value until componentDidMount
+    // so we can propagate any changes upstream
+    this.state = {};
+  }
 
-    this.state = {
-      value: this.getValue(props, context)
-    };
+  componentDidMount() {
+    this.setValue(this.getValue(this.props, this.context));
```

Comment:
this is a little gross, but it solves the problem pretty well without introducing any additional complexity

Reply:
Yeah I find the data flow a bit confusing overall, this seems to be fine though

I also still think using OOP is very un-react-like, and will not lend itself well to future maintenance, especially for those that are more familiar w/ react patterns

Reply:
I think this is where OOP is the strongest, and javascript users should learn the pattern ;)

aside, uniflow feels like it'd be overly complex here, and all stores are globalized, so you'd effectively be just passing around callbacks (actions/dispatcher) anyways